### ISSUE TITLE
minimega: add generic internal vm search

### ISSUE THREAD

**Issue by djfritz** [sandia-minimega/minimega#431]
We have a handful of internal functions to find vms by related fields (like by bridge, etc). It would be nice to have a generic findVM() by any field (including tags) that returns a slice of matched vms (local vms, we could have a globalFindVM() equivalent). Then we could replace the myriad APIs that require you to specify a tap name, a vm id, etc. with some unique VM search term (unique in that findVM() only returns a single vm). 

Things like the new QoS API would change from:

    qos add <interface> ...

to:

    qos add <vm search term> ...

As long as the search term hits on only one VM, we're good.

@floren @jcrussell 

Thoughts?

**Comment by djfritz** [comment]
Heck, we could add a vm: predicate that lets you tab complete just like the file: predicate.

**Comment by jcrussell** [comment]
I would prefer to wait on this until after the namespace branch gets merged. `findVm` works differently in the namespace branch (and better, IMO).

**Comment by djfritz** [comment]
Closing until after the namespace release. We can revisit this someday.
